libs/state/src/lib/core/operators/distinctUntilSomeChanged.ts

```diff
@@ -0,0 +1,173 @@
+import { MonoTypeOperatorFunction } from 'rxjs';
+import { distinctUntilChanged } from 'rxjs/operators';
+
+/**
+ * @internal
+ */
+export type CompareFn<T> = (x: T, y: T) => boolean;
+/**
+ * @internal
+ */
+export type KeyCompareMap<T extends object> = {
+  [K in keyof T]: CompareFn<T[K]>;
+};
+
+/**
+ * @description
+ *
```

Comment:
I would add information why the naming was picked and a link to MDN

Reply:
agree

Reply:
done